_specs/ecip-51-attack-protection-system.md

```diff
@@ -0,0 +1,157 @@
+The total difficulty is compressed simply by taking the 64 most significative bits.
+
+
+### Proof Size
+
+The size of a VisibilityProof message is variable. Normally it will contain a single Bitcoin block. However, it could contain more. We establish a maximum size of 10 Kilobytes per VisibilityProof proof. The additional bandwidth and storage space required by VisibilityProof messages is irrelevant to ETC.
+
+### Alarm Thresholds 
+
+We propose that for each window of 1000 blocks (with a delay of 256 blocks), the average visibility score is computed. If the visibility score decreases below 80% of the score of the previous interval, the node will enter a safe-mode and stop confirming transactions (in case of an exchange, it will not confirm an ETC deposit).
+
+Because RSK handles the construction of the merge-mining tag to be included in  Bitcoin blocks, the ETC community will be able to detect immediately if the Bitcoin network is producing VisibilityProof for a bock different from the ETC honest bestchain. Such event can trigger automatic notifications to exchanges and other economic actors.
+
+### Block Format
+
+To avoid changing the ETC block header format, the VisibilityProof proof can be included as an uncle, with a specil magic prefix to distiguish it from normal uncles. It can also be added in the ExtraData field. Of course, a new special field visibilityProofs can be added to the ETC block header.
+
+<< The exact format will be specified later >>
+
+### Inclusion of VisibilityProof Proofs
+
+On a first version, EtHash miners will be the only ones that will have the possibility to add VisibilityProofs to their blocks.
+
+In the future, the ETC network can propagate VisibilityProofs messages and we can let any user participate and collect fees from this system. This requires no changes to consensus. 
+
+### Rewards
+
+If a miner decides to add a VisibilityProof message in a slot, it will be rewarded by an extra subsidy. To prevent changing the ETC supply function, the extra subsidy will be obtained by diluting all remaining block subsidies by a fixed percentage. The exact amount is to be decided by the community but we propose this to be 10% of the current subsidy, starting at 1% and increasing until 10% during a period of 3 months. This will allow for EtHash miners to adapt and provides enough incentives for them to start using the RSK UMM system to obtain VisibilityProofs for their blocks.
```

Comment:
What if a miner does not include a proof, would that reward be lost or burnt?

Reply:
"lost" and "burnt" is the same. Maybe you mean accumulated (so the next miners gets a higher incentive) or burt. We (the authors) can define that, but I think this should be up to the ETC community to decide.